lib/code_review_master/data.rb

```diff
@@ -0,0 +1,144 @@
+  class Data
+    def initialize(channel, slack_username)
+      @channel = channel
+      @slack_username = slack_username
+    end
+
+    def add_reviewer!(segment)
+      begin
+        user = CodeReviewMaster::MessageSegment.new(segment).to_user
+        return user if reviewers.map(&:values).flatten.any? { |name| user.to_h.values.include?(name) }
+        return user if (config['reviewers'] << user.to_h && save!)
+      rescue CodeReviewMaster::MessageSegment::ParseError => e
+        SlackbotFrd::Log.debug(
+          "Failed to parse user segnment: #{segment}"
+        )
+        false
+      end
+    end
+
+    def assign_reviewer!(change_id)
+      dup_reviewers = reviewers.dup
+      slack_user_index = dup_reviewers.index do |reviewer|
+        reviewer['slack'] == slack_username
+      end
+      if slack_user_index != 0
+        assigned = dup_reviewers.shift
+      else
+        assigned = dup_reviewers.slice!(1)
+      end
+
+      dup_reviewers << assigned
+      self.reviewers = dup_reviewers
+      save!
+      CodeReviewMaster::User.new(assigned)
+    end
+
+    def clear_reviewers!
+      self.reviewers = []
+      save!
+    end
+
+    def persist_to_gerrit!(change_id:, assigned:)
+      if change_api
+        begin
+          change_api.add_reviewer(
+            id: change_id,
+            account_id: assigned.gerrit_username
+          )
+        rescue StandardError => e
+          SlackbotFrd::Log.warn(
+            "Error encountered sending reviewer to gerrit #{change_id}'.  " \
+            "Message: #{e.message}.\n#{e}"
+          )
+          false
+        end
+      else
+        SlackbotFrd::Log.warn(
+          "Gerrit is not configured."
+        )
+        false
+      end
+    end
+
+    def remove_reviewer!(name)
+      SlackbotFrd::Log.warn(
+        "Remove reviewer name: #{name} " \
+        "#{reviewers.map(&:values).flatten}"
+      )
+      return true unless reviewers.map(&:values).flatten.include?(name)
+      return true if config['reviewers'].delete_if do |reviewer|
+        reviewer.values.include?(name)
+      end && save!
+      false
+    end
+
+    def reviewers
+      config['reviewers']
+    end
+
+    def users
+      reviewers.map {|reviewer| CodeReviewMaster::User.new(reviewer) }
+    end
+
+    private
+    attr_reader :channel, :slack_username
+
+    def change_api
+      return nil unless gerrit_configured?
+
+      @_change_api ||= Gerrit::Change.new(
+        username: $slackbotfrd_conf['gerrit_username'],
+        password: $slackbotfrd_conf['gerrit_password']
+      )
+    end
+
+    def config
+      unless @_config
+        item = db.get_item(
+          table: table_name,
+          primary: channel
+        ).item
+        @_config = (item || {
+          'reviewers' => []
+        })
+      end
+      @_config
+    end
+
+    def db
+      unless @_db
+        @_db = DynamoDB.new(botname: 'angelbot')
+        @_db.create_table(table_name, if_not_exist: true)
```

Comment:
I know that this is commented out in the other class that creates a table (or did at one point). Is that what I should do here as well?

Reply:
Yeah, we'll need to create the table manually from the looks of it. The `botname` argument in the initialization should probably be `codereviewmaster` though.

Reply:
I am fine with updating the bot name. The other db integration just used the general `angelbot` so I was just following that. Make sense to differentiate though.